menuinst/platforms/linux.py

```diff
@@ -374,20 +374,34 @@ def _glob_pattern_for_mime_type(
 
         subcommand = "install" if install else "uninstall"
         # Install the XML file and register it as default for our app
+        use_fallback = False
         try:
             with TemporaryDirectory() as tmp:
                 with open(os.path.join(tmp, os.path.basename(xml_path)), "wb") as f:
                     tree.write(f, encoding="UTF-8", xml_declaration=True)
-                logged_run(
-                    ["xdg-mime", subcommand, "--mode", self.menu.mode, "--novendor", f.name],
-                    check=True,
-                )
+
+                xdg_mime = shutil.which("xdg-mime")
+                if xdg_mime:
+                    logged_run(
+                        [xdg_mime, subcommand, "--mode", self.menu.mode, "--novendor", f.name],
+                        check=True,
+                    )
+                else:
+                    name = self.render_key("name")
+                    log.warning(
+                        f"Unable to un/register MIME type '{glob_pattern}' "
+                        f"for Desktop Entry of name '{name}', "
+                        "'xdg-mime' is not present on the system.'"
+                    )
```

Comment:
f-strings in logging are generally discouraged. Python's `logging` module has built-in formatting functionality that is only called when needed: https://docs.astral.sh/ruff/rules/logging-f-string/